src/main/java/htsjdk/tribble/util/ParsingUtils.java

```diff
@@ -429,10 +430,16 @@ public static boolean resourceExists(String resource) throws IOException{
      * @return
      */
     public static URLHelper getURLHelper(URL url) {
-        try {
-            return getURLHelper(urlHelperClass, url);
-        } catch (Exception e) {
-            return getURLHelper(defaultUrlHelperClass, url);
+
+        if(urlHelperFactory != null) {
+            return urlHelperFactory.getHelper(url);
+        }
+        else {
```

Comment:
It would be good to add a comment here saying that this path is necessary for backwards compatibility with the old  deprecated code path.